src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -254,6 +257,26 @@ def app_post_deploy_hooks(self) -> Optional[List[ApplicationPostDeployHook]]:
     def debug_mode(self) -> bool:
         return self.na_project.debug_mode
 
+    @contextmanager
+    def use_warehouse(self, new_wh: Optional[str]):
```

Comment:
This is similar to the `use_role` contextmanager we also have, but that is part of the parent `SqlExecutionMixin` class as it could be widely used. But the `use_warehouse` is only being used in this new change. Hence keeping it here. But open to:
1. Bumping this up to the parent
2. Creating a generic context manager `use_object` similar to the `self.use` method on line 273.